ansible_collections/arista/cvp/docs/how-to/v3/cv_tag_v3.md

```diff
@@ -0,0 +1,129 @@
+# Execute / Cancel tasks on CloudVision
+
+__cv_tag_v3__ manage tasks on CloudVision:
+
+- Create tag on CloudVision
+- Assign tag on CloudVision
+- Unassign tag on CloudVision
+- Delete tag on CloudVision
+
+## Inputs
+
+The documentation is available in the [module section](../../modules/cv_tag_v3.rst.md).
+
+### Input variables
+
+- list of tags
+- state (`assign` or `unassign`)
+- mode (`create` or `delete`)
+- auto_create (`True` or `False`). Default is `True`
+
```

Comment:
I think we have too many inputs variable for that module. I did some tests and:
*  state: **assign** can be used only with mode: **create**.

```
TASK [create tags] ************************************************************************************************************************************************************************************************************
fatal: [cv_server]: FAILED! => {"changed": false, "msg": "Error, state cannot be 'unassign' when in 'create' mode"}

```

* state: **unassign** can be used only with mode: **delete**
```
TASK [create tags] ************************************************************************************************************************************************************************************************************
fatal: [cv_server]: FAILED! => {"changed": false, "msg": "Error, state cannot be 'assign' when in 'delete' mode"}
```

Maybe I'm missing something but this seems redudant. Could we not delete one of this field to simplify the module?

Reply:
`assign` is used when you need to assign an unassigned tag, likewise for `unassign`. `create` and `delete` is used to create or delete a tag. Sometimes, you only need to create a tag and not assign it or just unassign a tag and leave it as it is. That's why we separated the functions into `state` and `mode` category.

Reply:
Understood, I'm just wondering if there is really a use-case of being able to do the following scenarios:
* Create a tag without assigning it anywhere
* Unassign a tag without deleting it (if not used anywhere else)

If there is no real use case, we can maybe hide that logic and do something like: 
* While assigning, check if the tag exists. If the tag doesn't exist, create it before assign it. 
* While unassigning, check if the tag is used somewhere else. If the tag is not used on any other device, delete it. 

I also tried to delete a tag that is used on another device, and that doesn't seem to throw any error or do anything. 

Thus, maybe we can remove one of the 2 fields and simplify the usage of the module. In the end, Ansible is more about how we would like the system to be, and not focus on those kinds of details. 

My 2 cents on this :)